src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -35,20 +35,35 @@ use super::promote_consts::{self, Candidate, TempState};
 /// What kind of item we are in.
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
 enum Mode {
-    Const,
+    /// `static` item
     Static,
+    /// `static mut` item
     StaticMut,
+    /// `const fn`
     ConstFn,
-    Fn
+    /// `const` items and anonymous constants (e.g. in array lengths)
+    OtherConst,
+    /// other `fn`
+    NonConstFn,
+}
+
+impl Mode {
+    /// Determine whether we are running in "const context".  "const context" refers
```

Comment:
```suggestion
    /// Determine whether we are running in "const context". A "const context" refers
```

Reply:
I don't think this is correct?

Reply:
I believe "const context" is a countable singular noun. Those should be prefixed by an indefinite article whereas if you said "const contexts" then you wouldn't need an article.

Reply:
Not sure about the grammatical terms here, but I also think "in a const context" is wrong. It's "in const context". So maybe it's not countable or so.

Reply:
"In a const context" is right. =P  One const context; two const contexts; many const contexts. See also: https://doc.rust-lang.org/nightly/reference/const_eval.html#const-context

Reply:
> "In a const context" is right. =P

I don't think it is. You can't be "in two const contexts", that makes no sense.

One also says ["in context"](https://www.merriam-webster.com/dictionary/in%20context), not "in a context".

Reply:
I have to second @Centril here, sorry. The determiner is needed here. "context" can occasionally be a mass noun, but we're talking of actual individual contexts here, so it just sounds wrong without a determiner. :-)

Reply:
We're not really referring to an individual context. It's not like this is like a varibale context in a type system or so. This is just a state to be in. "I am in pain". "This expression is in const context".

I should probably just avoid the term entirely... everybody seems confused by it.^^